src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -25,7 +25,7 @@ namespace detail {
 template <typename IntegrandType>
 double integrand(const double x, void* const params) {
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-  const auto function = reinterpret_cast<IntegrandType*>(params);
+  auto* const function = reinterpret_cast<IntegrandType*>(params);
```

Comment:
So I understand, please explain this change?

Reply:
@wthrowe this was a clang-tidy change. I remember you commented on the original code, could you weigh in if this is correct?

Reply:
It looks fine.  It makes it obvious that `function` is a pointer, which is nice.